lib/tom_queue/worker_supervisor.rb

```diff
@@ -0,0 +1,67 @@
+module TomQueue
+# class WorkerPool?
+  class WorkerSupervisor
+
+    attr_reader :processes
+
+    def initialize(processes = default_processes)
+      @processes = processes
+      setup_signal_handling
+    end
+
+    def default_processes
+      # TODO: deferred worker
+      {
+        "worker1" => -> { TomQueue::Worker.new.start },
```

Comment:
I'd be inclined to offer a `supervisor.supervise(klass, as: "name")` method, so the caller can establish what processes actually get started up ?